### PR TITLE
Handle commenting of multi-line SQL in getDownQueryComments

### PR DESCRIPTION
**Pull request by etrepum**
### Description

I noticed that the down migration placeholder generated by `getDownQueryComments` does not handle commenting of multi-line SQL, which can result in a down.sql that looks like this:

```sql
-- Could not auto-generate a down migration.
-- Please write an appropriate down migration for the SQL below:
-- alter table "public"."locations" add column "google_maps_url" text
null;
```

### Changelog

- [ ] `CHANGELOG.md` is updated with user-facing content relevant to this PR. If no changelog is required, then add the `no-changelog-required` label.

### Affected components

- [ ] Server
- [X] Console
- [ ] CLI
- [ ] Docs
- [ ] Community Content
- [ ] Build System
- [ ] Tests
- [ ] Other (list it)

#### Catalog upgrade

Does this PR change Hasura Catalog version?
- [X] No
- [ ] Yes
  - [ ] Updated docs with SQL for downgrading the catalog 

#### Metadata

Does this PR add a new Metadata feature?
- [X] No
- [ ] Yes
  - Does `run_sql` auto manages the new metadata through schema diffing?
    - [ ] Yes
    - [ ] Not required
  - Does `run_sql` auto manages the definitions of metadata on renaming?
    - [ ] Yes
    - [ ] Not required
  - Does `export_metadata`/`replace_metadata` supports the new metadata added?
    - [ ] Yes
    - [ ] Not required

#### GraphQL
- [X] No new GraphQL schema is generated
- [ ] New GraphQL schema is being generated:
   - [ ] New types and typenames are correlated
   
   

#### Breaking changes

- [X] No Breaking changes
- [ ] There are breaking changes:

  1. Metadata API

     Existing `query` types:
     - [ ] Modify `args` payload which is not backward compatible
     - [ ] Behavioural change of the API
     - [ ] Change in response `JSON` schema
     - [ ] Change in error code
     

  2. GraphQL API

     Schema Generation:
     
     
     - [ ] Change in any `NamedType`
     - [ ] Change in table field names
     

     Schema Resolve:-
     
     - [ ] Change in treatment of `null` value for any input fields 
     

  3. Logging

     - [ ] Log `JSON` schema has changed
     - [ ] Log `type` names have changed

### CODE REVIEW

File: console/src/utils/migration/utils.ts [hunk 1]
```diff
@@ -349,13 +349,11 @@ export const getDownQueryComments = (
...
-    comment = upqueries.reduce(
-      (acc, i) => `${acc}
--- ${i.args.sql}`,
-      comment
-    );
+    const comment = [
+      "Could not auto-generate a down migration.",
+      "Please write an appropriate down migration for the SQL below:",
+      ...upqueries.map((i) => i.args.sql)
+    ].join("\n").replace(/\r?\n/g, "$1-- ");
```
Comment (soorajshankar) [inline console/src/utils/migration/utils.ts:356]: could you try this one?
small changes on the regex to make sure all carrier returns and line breaks are captured + commented out the first line directly.
```suggestion
    const comment = [
      '-- Could not auto-generate a down migration.',
      'Please write an appropriate down migration for the SQL below:',
      ...upqueries.map(i => i.args.sql),
    ]
      .join('\n')
      .replace(/(\r\n)+|\r+|\n+/g, '\n-- ');
```

Comment (soorajshankar) [reply console/src/utils/migration/utils.ts:356]: with the above suggestion, I was able to generate the following

```sql
-- Could not auto-generate a down migration.
-- Please write an appropriate down migration for the SQL below:
-- CREATE OR REPLACE VIEW author_publics_2 AS
--   SELECT id, name,city, email
--     FROM author;
```

**Review by soorajshankar — CHANGES_REQUESTED** [review]
Hello @etrepum,

Thank you for raising this PR.

I was trying to make a multiline SQL migration, but it seem to be generating the following string.

```
Could not auto-generate a down migration.$1-- Please write an appropriate down migration for the SQL below:$1-- CREATE OR REPLACE VIEW author_public AS$1--   SELECT id, name,city, email$1--     FROM author;
```

left a small suggestion, could you try that and update?

File: console/src/utils/migration/utils.ts (hunk 1 above)
Comment (etrepum) [reply console/src/utils/migration/utils.ts]: Bare carriage returns aren't used in the modern world, it's either `\n` or `\r\n`. I've revised and rebased the PR to ensure the first line is commented, newlines are normalized, lines do not have trailing spaces, and the file ends with a newline.

**Review by soorajshankar — APPROVED** [review]
code and functionality LGTM

**Pull request closed without merging**